core/meta/src/TClass.cxx

```diff
@@ -2713,6 +2713,9 @@ Int_t TClass::GetBaseClassOffset(const TClass *toBase, void *address, bool isDer
 
    if (this == toBase) return 0;
 
+   if (!fHasMultipleInheritance && nullptr == address &&
+       isDerivedObject && InheritsFrom(toBase)) return 0;
```

Comment:
Here I have a general question: why is address always `nullptr` in cases where we can return zero?